edgelet/edgelet-http/src/certificate_manager.rs

```diff
@@ -0,0 +1,175 @@
+use std::str;
+use std::sync::{Arc, RwLock};
+
+use edgelet_core::crypto::{Certificate, CreateCertificate};
+use edgelet_core::{CertificateIssuer, CertificateProperties, CertificateType};
+use failure::ResultExt;
+
+pub use crate::error::{Error, ErrorKind};
+
+const IOTEDGED_VALIDITY: u64 = 7_776_000; // 90 days
+const IOTEDGED_TLS_COMMONNAME: &str = "iotedge tls";
+
+#[derive(Clone)]
+pub struct CertificateManager<C: CreateCertificate + Clone> {
+    certificate: Arc<RwLock<String>>,
+    crypto: C,
+}
+
+impl<C: CreateCertificate + Clone> CertificateManager<C> {
+    pub fn new(crypto_struct: C) -> Result<Self, Error> {
+        Ok(CertificateManager {
+            certificate: Arc::new(RwLock::new("".to_string())),
+            crypto: crypto_struct,
+        })
+    }
+
+    pub fn get_certificate(&self) -> Result<String, Error> {
+        // First, try to directly read
+        {
+            let cert = self
+                .certificate
+                .read()
+                .expect("Locking the certificate for read failed.")
+                .to_string();
```

Comment:
Do the `.to_string()` inside the `if`. No point wasting cycles converting an empty `&str` to an empty `String` and then throwing it away.